Uni_life_tracker/app/src/main/java/Database/UserInfoDatabaseHelper.java

```diff
@@ -0,0 +1,168 @@
+package Database;
+
+import android.annotation.SuppressLint;
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+
+public class UserInfoDatabaseHelper extends SQLiteOpenHelper {
+
+    private static final String createUserInfoTable = "CREATE TABLE userinfo(NAME TEXT, " +
+            "UNIVERSITY TEXT, EMAIL TEXT, PASSWORD TEXT)";
+    private SQLiteDatabase db;
+
+    public UserInfoDatabaseHelper(Context context) {
+        super(context, "userInfoDatabase", null, 1);
+    }
+
+    @Override
+    public void onCreate(SQLiteDatabase db) {
+        db.execSQL(createUserInfoTable);
+    }
+
+    @Override
+    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+        db.execSQL("DROP TABLE IF EXISTS userinfo");
+        onCreate(db);
+    }
+
+
+    /**
+     * Open the database for reading or writing
+     */
+    public void openDatabase() {
+        db = getWritableDatabase();
+    }
+
+    /**
+     * Close the database
+     */
+    public void closeDatabase() {
+        db.close();
+    }
+
+    /**
+     * Insert User Info into the database if initially empty
+     *
+     * @param name The user's name to be inserted
+     * @param uni The user's university to be inserted
+     * @param email The user's email to be inserted
+     * @param password The user's password to be inserted
+     */
+    public void insertUserInfo(String name, String uni, String email, String password){
+        @SuppressLint("Recycle") Cursor cur = db.rawQuery("SELECT * FROM userinfo",
+                null);
+        if (!cur.moveToFirst()) {
+            // if database is empty then add the User's info into the database
+            ContentValues cv = new ContentValues();
```

Comment:
I'm wondering if using the keyword INSERT from SQL would be better than using put? not sure honestly

Reply:
I was looking into this and found that using methods that take in parameters that make up a query is better than just straight up using the raw query. This also applies to the above comment. This link explains why this way is better than just writing down the queries as strings.

https://stackoverflow.com/questions/43446062/what-is-the-difference-between-inserting-data-using-contentvalues-and-raw-sql-in